src/test/lib/memory/tracking_memory_resource_test.cpp

```diff
@@ -0,0 +1,61 @@
+#include "base_test.hpp"
+#include "memory/tracking_memory_resource.hpp"
+
+namespace opossum {
+
+class TrackingMemoryResourceTest : public BaseTest {
+ protected:
+  TrackingMemoryResource tracking_memory_resource;
+};
+
+TEST_F(TrackingMemoryResourceTest, RecordAllocations) {
+  ASSERT_EQ(tracking_memory_resource.memory_timeseries().size(), 0);
+
+  tracking_memory_resource.allocate(sizeof(int));
+  ASSERT_EQ(tracking_memory_resource.memory_timeseries().size(), 1);
+  const auto& [recorded_time_int, recorded_amount_int] = tracking_memory_resource.memory_timeseries()[0];
+  EXPECT_EQ(recorded_amount_int, sizeof(int));
+  EXPECT_LT(recorded_time_int, std::chrono::system_clock::now());
+  EXPECT_GT(recorded_time_int, std::chrono::system_clock::now() - std::chrono::milliseconds(100));
+
+  tracking_memory_resource.allocate(sizeof(float) * 2);
+  ASSERT_EQ(tracking_memory_resource.memory_timeseries().size(), 2);
+  const auto& [recorded_time_float, recorded_amount_float] = tracking_memory_resource.memory_timeseries()[1];
+  EXPECT_EQ(recorded_amount_float, sizeof(float) * 2);
+  EXPECT_LT(recorded_time_float, std::chrono::system_clock::now());
+  EXPECT_GT(recorded_time_float, std::chrono::system_clock::now() - std::chrono::milliseconds(100));
+}
+
+TEST_F(TrackingMemoryResourceTest, RecordDeallocations) {
+  ASSERT_EQ(tracking_memory_resource.memory_timeseries().size(), 0);
+
+  auto mem_ptr_1 = std::malloc(10);
```

Comment:
I would argue that `ptr` is a default abbreviation, which can also be found in the std lib (see shared_ptr, unique_ptr). Nevertheless, I would try to avoid the `mem` abbreviation here.